build/templates/flutter-diagnostics.yml

```diff
@@ -0,0 +1,15 @@
+parameters:
+- name: projectDirectory
+  type: string
+  default: '$(Build.SourcesDirectory)/src/app'
+- name: condition
+  type: string
+  default: always()
+
+steps:
+  - task: FlutterCommand@0
```

Comment:
what does it do exactly?

Reply:
Run if an important Flutter-related command fails, which takes about 2 minutes. It's a small optimization. And since the condition can't be applied to a template, I had to do it like that.

Reply:
[√] Flutter (Channel stable, 3.19.6, on Microsoft Windows [Version 10.0.22631.3593], locale en-CA)
    • Flutter version 3.19.6 on channel stable at C:\Dev\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 54e66469a9 (6 weeks ago), 2024-04-17 13:08:03 -0700
    • Engine revision c4cd48e186
    • Dart version 3.3.4
    • DevTools version 2.31.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\thomas.labrecque\AppData\Local\Android\sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Professional 2022 17.10.1)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Professional
    • Visual Studio Professional 2022 version 17.10.34928.147
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code, 64-bit edition (version 1.89.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.90.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.3593]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 125.0.6422.114
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 125.0.2535.67

[√] Network resources
    • All expected network resources are available.

• No issues found!

Reply:
Required information to debug potential issues.

Reply:
oh some sort of flutter doctor. Nice